src/legacy/core_plugins/interpreter/public/functions/visualization.js

```diff
@@ -0,0 +1,147 @@
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { get } from 'lodash';
+import { i18n } from '@kbn/i18n';
+import chrome from 'ui/chrome';
+import { VisRequestHandlersRegistryProvider as RequestHandlersProvider } from 'ui/registry/vis_request_handlers';
+import { VisResponseHandlersRegistryProvider as ResponseHandlerProvider } from 'ui/registry/vis_response_handlers';
+import { VisTypesRegistryProvider } from 'ui/registry/vis_types';
+import { IndexPatternsProvider } from 'ui/index_patterns';
+import { FilterBarQueryFilterProvider } from 'ui/filter_bar/query_filter';
+import { PersistedState } from 'ui/persisted_state';
+
+function getHandler(from, type) {
+  if (typeof type === 'function') {
+    return type;
+  }
+  if (type === 'courier' || type === 'none') {
+    return null;
+  }
+  const handlerDesc = from.find(handler => handler.name === type);
+  if (!handlerDesc) {
+    throw new Error(`Could not find handler "${type}".`);
+  }
+  return handlerDesc.handler;
+}
+
+export default () => ({
+  name: 'visualization',
+  type: 'render',
+  context: {
+    types: [],
+  },
+  help: i18n.translate('common.core_plugins.interpreter.public.functions.visualization.help', {
+    defaultMessage: 'A simple visualization'
+  }),
+  args: {
+    index: {
+      types: ['string', 'null'],
+      default: null,
+    },
+    metricsAtAllLevels: {
+      types: ['boolean'],
+      default: false,
+    },
+    partialRows: {
+      types: ['boolean'],
+      default: false,
+    },
+    type: {
+      types: ['string'],
+      default: '',
+    },
+    schemas: {
+      types: ['string'],
+      default: '"{}"',
+    },
+    visConfig: {
+      types: ['string'],
+      default: '"{}"',
+    },
+    uiState: {
+      types: ['string'],
+      default: '"{}"',
+    }
+  },
+  async fn(context, args, handlers) {
+    const $injector = await chrome.dangerouslyGetActiveInjector();
+    const Private = $injector.get('Private');
+    const requestHandlers = Private(RequestHandlersProvider);
+    const responseHandlers = Private(ResponseHandlerProvider);
+    const visTypes = Private(VisTypesRegistryProvider);
+    const indexPatterns = Private(IndexPatternsProvider);
+    const queryFilter = Private(FilterBarQueryFilterProvider);
+
+    const visConfigParams = JSON.parse(args.visConfig);
```

Comment:
Any reason we shouldn't catch and throw a helpful message if the argument isn't successfully parsed?

Reply:
for now we don't give any additional help from within functions, if they fail we will catch the error at the executor level and just report something went wrong.